server/api/views.js

```diff
@@ -1,24 +1,26 @@
 // @flow
 import Router from 'koa-router';
-import httpErrors from 'http-errors';
 import auth from './middlewares/authentication';
 import { presentView } from '../presenters';
 import { View, Document } from '../models';
+import policy from '../policies';
 
+const { authorize } = policy;
 const router = new Router();
 
 router.post('views.list', auth(), async ctx => {
   const { id } = ctx.body;
   ctx.assertPresent(id, 'id is required');
 
+  const user = ctx.state.user;
+  const document = await Document.findById(id);
+  authorize(user, 'read', document);
```

Comment:
Previously there was no check here - you could record a "view" against any document in the db if you know the `id`. `views.list` would then happily return that document.